lib/staggered_dslash.cu

```diff
@@ -21,53 +21,80 @@ namespace quda {
 namespace quda {
 
 
+//   template <int dir, typename I>
+//   __device__ __host__ inline int mwghostFaceIndex(const int x[], const I X[], int dim, int nFace) {
+//     int index = 0;
+//     switch(dim) {
+//     case 0:
+//       switch(dir) {
+//       case 0:
+//   index = ((x[0]+nFace-1)*X[4]*X[3]*X[2]*X[1] + x[4]*X[3]*X[2]*X[1] + x[3]*(X[2]*X[1])+x[2]*X[1] + x[1])>>1;
+//     // index = ((x[0])*X[4]*X[3]*X[2]*X[1] + x[4]*X[3]*X[2]*X[1] + x[3]*(X[2]*X[1])+x[2]*X[1] + x[1])>>1;
+//   break;
+//       case 1:
+//   index = ((x[0]-X[0]+nFace)*X[4]*X[3]*X[2]*X[1] + x[4]*X[3]*X[2]*X[1] + x[3]*(X[2]*X[1]) + x[2]*X[1] + x[1])>>1;
+//   break;
+//  }  
+//   }
+//  return index;
+// }
+
+ /**
+     Compute the checkerboarded index into the ghost field
+     corresponding to full (local) site index x[]
+     @param x local site
+     @param X local lattice dimensions
+     @param dim dimension
+     @param depth of ghost
+  */
   template <int dir, typename I>
   __device__ __host__ inline int mwghostFaceIndex(const int x[], const I X[], int dim, int nFace) {
     int index = 0;
     switch(dim) {
-      case 0:
+    case 0:
       switch(dir) {
-        case 0:
-        index = ((x[0]+nFace-1)*X[4]*X[3]*X[2]*X[1] + x[4]*X[3]*X[2]*X[1] + x[3]*(X[2]*X[1])+x[2]*X[1] + x[1])>>1;
-    // index = ((x[0])*X[4]*X[3]*X[2]*X[1] + x[4]*X[3]*X[2]*X[1] + x[3]*(X[2]*X[1])+x[2]*X[1] + x[1])>>1;
-        break;
-        case 1:
-        index = ((x[0]-X[0]+nFace)*X[4]*X[3]*X[2]*X[1] + x[4]*X[3]*X[2]*X[1] + x[3]*(X[2]*X[1]) + x[2]*X[1] + x[1])>>1;
-        break;
-      }  
+      case 0:
+ index = ((x[0]+nFace-1)*X[4]*X[3]*X[2]*X[1] + x[4]*X[3]*X[2]*X[1] + x[3]*(X[2]*X[1])+x[2]*X[1] + x[1])>>1;
```

Comment:
This may get addressed in a later pull, but just in case, @mathiaswagner can math like this which includes `nFace` get put into a helper routine in `dslash_helper`? Routines like this for other ops live there, iirc

Reply:
Found it, all good!